common/client_flags.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2021 The Rode Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import "flag"
+
+func SetupRodeClientFlags(flags *flag.FlagSet) *ClientConfig {
+	conf := &ClientConfig{
+		Rode:      &RodeClientConfig{},
+		JWTAuth:   &JWTAuthConfig{},
+		BasicAuth: &BasicAuthConfig{},
+	}
+
+	flags.StringVar(&conf.Rode.Host, "rode-host", "rode:50051", "the host to use to connect to rode")
+	flags.BoolVar(&conf.Rode.DisableTransportSecurity, "rode-insecure-disable-transport-security", false, "when set, the connection to rode will not use transport security")
+
+	flags.StringVar(&conf.JWTAuth.ClientID, "jwt-client-id", "", "the client ID to use when requesting a JWT via the client_credentials grant")
+	flags.StringVar(&conf.JWTAuth.ClientSecret, "jwt-client-secret", "", "the client secret to use when requesting a JWT via the client_credentials grant")
+	flags.StringVar(&conf.JWTAuth.TokenURL, "jwt-token-url", "", "the URL to use to retrieve an access token via the client_credentials grant")
```

Comment:
I was stuck between using the `jwt` prefix or `oauth`, which would more accurately describe the flow in which these config values are used.  open to feedback

Reply:
IIRC the access token in OAuth 2.0 doesn't have a specified format; even though commonly it's a JWT it can be an encrypted/signed blob too. So `oauth` might be a better fit.

Reply:
To be clear, the reason I ended up going with `jwt` here is because that is how we treat the access token once the rode server receives it.  The command line flags for rode re-enforce this by allowing you to specify an authorized issued and a path to find the role claims.  so I thought `jwt` was more consistent, despite the fact that all the client is doing here is `oauth`.

Reply:
Tangential to this, but IMO we should rename the Rode flags, JWT auth could mean a number of things but we expect an OIDC server for discovery and JWKS at the moment.